src/kinetics/Reaction.cpp

```diff
@@ -37,6 +37,21 @@ Reaction::Reaction(const Composition& reactants_,
     , m_third_body(tbody_)
 {
     setRate(rate_);
+
+    // ensure safe serialization
+    size_t count = 0;
+    for (const auto& reac : reactants) {
+        if (products.count(reac.first)) {
+            count++;
+        }
+    }
```

Comment:
Seems like this could be a little simpler as:

```suggestion
    bool count = false;
    for (const auto& reac : reactants) {
        if (products.count(reac.first)) {
            count = true;
            break;
        }
    }
```